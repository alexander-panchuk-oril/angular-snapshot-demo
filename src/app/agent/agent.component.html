<div class="card">
  <ng-container *ngIf="layout === 'rtl'; then btn; else title"></ng-container>
  <ng-container *ngIf="layout === 'rtl'; then title; else btn"></ng-container>
</div>

<ng-template #title>
  <h4>{{ agent.codeName }}</h4>
</ng-template>
<ng-template #btn>
  <button class="primary">Hire</button>
</ng-template>
