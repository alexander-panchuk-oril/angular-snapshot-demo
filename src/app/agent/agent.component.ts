import { Component, Input, OnInit } from '@angular/core';
import { ISecretAgent } from '../shared/interfaces/ISecretAgent.interface';

@Component({
  selector: 'app-agent',
  templateUrl: './agent.component.html',
  styleUrls: ['./agent.component.scss']
})
export class AgentComponent implements OnInit {

  @Input()
  public layout: 'ltr' | 'rtl';

  @Input()
  public agent: ISecretAgent;

  constructor() { }

  ngOnInit(): void {
  }

}
