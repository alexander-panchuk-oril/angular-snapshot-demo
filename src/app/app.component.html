<section>
    <div class="container">
        <h1>Secret Agents</h1>
        <button class="secondary" (click)="swapLayout()">
             Swap Layout
        </button>
        <button class="secondary" (click)="sortByName()">
             Sort By Name
        </button>
        <app-agent *ngFor="let agent of agents" [agent]="agent" [layout]="layout"></app-agent>
    </div>
</section>