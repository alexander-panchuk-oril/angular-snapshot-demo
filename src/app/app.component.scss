section {
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    .container {
        max-width: 480px;
        text-align: center;
    }
}
h1 {
    text-align: center;
}
button {
    margin-bottom: 1rem;
    margin-left: 1rem;
    &:first-of-type {
        margin-left: 0;
    }
}
app-agent {
    display: inline-block;
    margin-bottom: 1rem;
    width: 100%;
}
