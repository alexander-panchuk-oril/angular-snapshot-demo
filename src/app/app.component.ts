import { Component, OnInit } from '@angular/core';
import { ISecretAgent } from './shared/interfaces/ISecretAgent.interface';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit {
  public layout: 'rtl' | 'ltr' = 'ltr';
  public agents: ISecretAgent[] = [
    {
      codeName: 'Velvet Thunder',
      agency: 'Alpha Protocol'
    },
    {
      codeName: 'Wet Blanket',
      agency: 'F.I.R.M.'
    },
    {
      codeName: 'Soft Hummer',
      agency: 'Section 31'
    },
    {
      codeName: 'Sidewinter',
      agency: 'DXS'
    },
    {
      codeName: 'Dizzy Rider',
      agency: 'The Agency'
    }
  ];

  ngOnInit() {
  }

  public swapLayout() {
    this.layout = this.layout === 'rtl' ? 'ltr' : 'rtl';
  }

  public sortByName(): ISecretAgent[] {

    return this.agents.sort((a, b) => a.codeName.localeCompare(b.codeName));
  }
}
