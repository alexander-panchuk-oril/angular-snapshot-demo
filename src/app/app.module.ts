import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';
import { AgentComponent } from './agent/agent.component';

@NgModule({
  declarations: [
    AppComponent,
    AgentComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
