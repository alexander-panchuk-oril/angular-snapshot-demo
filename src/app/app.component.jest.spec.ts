import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { AgentComponent } from './agent/agent.component';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        FontAwesomeModule
      ],
      declarations: [
        AppComponent,
        AgentComponent
      ],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should sort agents by name', () => {
    const sortedAgents = fixture.componentInstance.sortByName();
    expect(sortedAgents).toEqual([
      {
        codeName: 'Dizzy Rider',
        agency: 'The Agency'
      },
      {
        codeName: 'Sidewinter',
        agency: 'DXS'
      },
      {
        codeName: 'Soft Hummer',
        agency: 'Section 31'
      },
      {
        codeName: 'Velvet Thunder',
        agency: 'Alpha Protocol'
      },
      {
        codeName: 'Wet Blanket',
        agency: 'F.I.R.M.'
      }
    ]);
  });

  it('[snapshot] should sort agents by name', () => {
    const sortedAgents = component.sortByName();
    expect(sortedAgents).toMatchSnapshot();
  });
});
