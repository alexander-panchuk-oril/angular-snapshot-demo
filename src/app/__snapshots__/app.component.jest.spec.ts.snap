// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent [snapshot] should sort agents by name 1`] = `
Array [
  Object {
    "agency": "The Agency",
    "codeName": "Dizzy Rider",
  },
  Object {
    "agency": "DXS",
    "codeName": "Sidewinter",
  },
  Object {
    "agency": "Section 31",
    "codeName": "Soft Hummer",
  },
  Object {
    "agency": "Alpha Protocol",
    "codeName": "Velvet Thunder",
  },
  Object {
    "agency": "F.I.R.M.",
    "codeName": "Wet Blanket",
  },
]
`;
